pkg/scm/downloaders/file/download.go

```diff
@@ -39,8 +39,20 @@ func (f *File) Download(config *api.Config) (*git.SourceInfo, error) {
 	if strings.HasPrefix(filepath.Clean(config.WorkingSourceDir), filepath.Clean(copySrc)) {
 		return nil, RecursiveCopyError{error: fmt.Errorf("recursive copy requested, source directory %q contains the target directory %q", copySrc, config.WorkingSourceDir)}
 	}
+
+	ignoreFiles := []string{}
+	absDst, err := filepath.Abs(config.WorkingSourceDir)
+	if err != nil {
+		return nil, err
+	}
+	if filepath.HasPrefix(absDst, copySrc) {
+		dir, _ := filepath.Split(config.WorkingSourceDir)
+		ignoreFiles = append(ignoreFiles, filepath.Clean(dir))
+	}
```

Comment:
we were attempting to handle this more bluntly on lines 39-41.  But obviously we missed converting the paths to absolute before comparing them.  I somewhat prefer fixing that logic though, and just considering this usage invalid (ie you always need to target a different directory for your output/Dockerfile)

Reply:
@bparees Oh right, how did I miss that… 

> considering this usage invalid (ie you always need to target a different directory for your output/Dockerfile)

I'm not sure why it should be invalid though ? or more accurately, I think in this case the error should "help" the user (telling it to specify another *output* directory).

Anyway, I'll update the conditions in lines 39-41 :wink:

Reply:
it doesn't have to be invalid, your approach could definitely work, i just find it simpler to reason about the idea that we should always create a clean destination directory rather than polluting the working directory which could just confuse the user.

Reply:
@bparees agreed :angel: :wink: :wine_glass: